src/cast_core/src/CrudActor.cpp

```diff
@@ -699,6 +699,65 @@ struct FindOneOperation : public BaseOperation {
     metrics::Operation _operation;
 };
 
+struct AggregateOperation : public BaseOperation {
+    struct Pipeline {
+        Pipeline(const Node& node, PhaseContext& context, ActorId id) {
+            if (!node.isSequence()) {
+                BOOST_THROW_EXCEPTION(InvalidConfigurationException("'Pipeline' must be an array"));
+            }
+            for (auto&& [_, stageNode] : node) {
+                DocumentGenerator stage = stageNode.to<DocumentGenerator>(context, id);
+                pipeline.append_stage(stage());
```

Comment:
This is being evaluated at constructor-time versus for each iteration. Result is you'll have a random pipeline but each invocation will use the same random pipeline. Is that intended? If not, you can instead collect a vector of `DocumentGenerator`s (one for each stage) and then construct the pipeline by calling the generators in the `run` block.

Reply:
Oh, great catch! I didn't think that deeply about _when_ the generator was being called when I implemented this. You're right that it makes more sense to do in the `run()` method. Done.